File: src/main/java/superstitio/relics/a_starter/StartWithSexToy.java
package superstitio.relics.a_starter;

import basemod.AutoAdd;
import com.megacrit.cardcrawl.actions.common.RelicAboveCreatureAction;
import com.megacrit.cardcrawl.dungeons.AbstractDungeon;
import superstitio.DataManager;
import superstitio.cards.general.TempCard.SexToy;
import superstitio.relics.AbstractLupaRelic;
import superstitio.utils.ActionUtility;

import static com.megacrit.cardcrawl.dungeons.AbstractDungeon.player;

@AutoAdd.Seen
public class StartWithSexToy extends AbstractLupaRelic {
    public static final String ID = DataManager.MakeTextID(StartWithSexToy.class);
    // 遗物类型
    private static final RelicTier RELIC_TIER = RelicTier.STARTER;
    // 点击音效
    private static final LandingSound LANDING_SOUND = LandingSound.FLAT;
    private static final int SexToyNum = 2;

    public StartWithSexToy() {
        super(ID, RELIC_TIER, LANDING_SOUND);
    }

    @Override
    public void atBattleStart() {
        this.flash();
        this.addToBot(new RelicAboveCreatureAction(AbstractDungeon.player, this));
        for (int i = 0; i < SexToyNum; i++) {
            ActionUtility.addToBot_applyPower(new superstitio.powers.SexToy(player, 1, SexToy.getRandomSexToyNameWithoutRng()));
        }
    }

    @Override
    public void updateDescriptionArgs() {
        setDescriptionArgs(SexToyNum);
    }
}

File: superstitioMain/java/superstitio/relics/a_starter/StartWithSexToy.kt
package superstitio.relics.a_starter

import basemod.AutoAdd.Seen
import com.megacrit.cardcrawl.actions.common.RelicAboveCreatureAction
import com.megacrit.cardcrawl.dungeons.AbstractDungeon
import superstitio.DataManager
import superstitio.powers.SexToy
import superstitio.relics.SuperstitioRelic
import superstitioapi.utils.ActionUtility
import superstitioapi.utils.setDescriptionArgs

@Seen
class StartWithSexToy : SuperstitioRelic(ID, RELIC_TIER, LANDING_SOUND)
{
    override fun atBattleStart()
    {
        this.flash()
        this.addToBot(RelicAboveCreatureAction(AbstractDungeon.player, this))
        for (i in 0 until SexToyNum)
        {
            ActionUtility.addToBot_applyPower(
                SexToy(
                    AbstractDungeon.player,
                    1,
                    superstitio.cards.general.TempCard.SexToy.Companion.getRandomSexToyNameWithoutRng()
                )
            )
        }
    }

    override fun updateDescriptionArgs()
    {
        setDescriptionArgs(SexToyNum)
    }

    companion object
    {
        val ID: String = DataManager.MakeTextID(StartWithSexToy::class.java)

        // 遗物类型
        private val RELIC_TIER = RelicTier.STARTER

        // 点击音效
        private val LANDING_SOUND = LandingSound.FLAT
        private const val SexToyNum = 2
    }
}